torch/csrc/distributed/autograd/context/dist_autograd_container.cpp

```diff
@@ -173,6 +180,22 @@ void DistAutogradContainer::eraseContextIdAndReset(int64_t context_id) {
   }
 }
 
+void DistAutogradContainer::cleanupContextWatchdog() {
+  std::this_thread::sleep_for(kContextTimeout);
+  while(!terminateWatchdog_.load()) {
+    std::lock_guard<std::mutex> guard(autograd_context_lock_);
+    for (auto& pair : autograd_context_) {
+      if (autograd_context_.find(pair.first) != autograd_context_.end()) { 
+        sendReleaseContextRpc(pair.first);
+        eraseContextIdAndReset(pair.first);
+        if (autograd_context_.empty()) {
```

Comment:
Since we grab the lock before the `for` loop, and iterate through all the contexts, removing them, it looks like this will always remove all the contexts, so this should always be true at the last iteration of the loop? If this is the intention, then we could remove this `if` check and have the return after the loop. 

On the other hand if the goal is to keep looping and clean up additional contexts created, then we could remove the return entirely. Although, I'm not sure if this is what we want, since it seems like it would cause serious lock contention...